File: src/test/java/org/skefir/data/CalendarTableColumn.java
package org.skefir.data;

import lombok.Getter;

public enum CalendarTableColumn implements DataTableColumn {
    TIME(    "Time,"),
    CURRENCY("Currency"),
    EVENT(   "Event"),
    ACTUAL(  "Actual"),
    FORECAST("Forecast"),
    PREVIOUS("Previous");

    @Getter
    final String title;

    CalendarTableColumn(String title) {
        this.title = title;
    }
}

File: src/test/kotlin/org/skefir/data/CalendarTableColumn.kt
package org.skefir.data



enum class CalendarTableColumn(private val title: String) : DataTableColumn {
    TIME(    "Time,"),
    CURRENCY("Currency"),
    EVENT(   "Event"),
    ACTUAL(  "Actual"),
    FORECAST("Forecast"),
    PREVIOUS("Previous");

    override fun getTitle(): String {
        return title
    }
    override fun getColumnByTitle(title: String): DataTableColumn = values().map{it.also { println("####${it.title} #-#${title}") }}.first { it.title == title }

}